profiler/profiler_test.go

```diff
@@ -202,14 +202,10 @@ func TestProfilerPassthrough(t *testing.T) {
 	}
 	p.run()
 	var bat batch
-loop:
-	for {
-		select {
-		case bat = <-p.out:
-			break loop
-		case <-time.After(500 * time.Millisecond):
-			t.Fatal("time expired")
-		}
+	select {
+	case bat = <-out:
+	case <-time.After(1000 * time.Millisecond):
```

Comment:
Maybe put a comment or TODO here in case there's time to improve it in the future.